Zend/zend_closures.c

```diff
@@ -75,14 +75,19 @@ ZEND_METHOD(Closure, __invoke) /* {{{ */
 /* }}} */
 
 static bool zend_valid_closure_binding(
-		zend_closure *closure, zval *newthis, zend_class_entry *scope) /* {{{ */
+		zend_closure *closure, zval **newthis_ptr, zend_class_entry *scope) /* {{{ */
 {
+	zval *newthis = *newthis_ptr;
 	zend_function *func = &closure->func;
 	bool is_fake_closure = (func->common.fn_flags & ZEND_ACC_FAKE_CLOSURE) != 0;
 	if (newthis) {
 		if (func->common.fn_flags & ZEND_ACC_STATIC) {
-			zend_error(E_WARNING, "Cannot bind an instance to a static closure, this will be an error in PHP 9");
-			return 0;
+			// FIXME: Restrict this workaround to implicitly static closures?
+			// Will need an additional fn_flag.
```

Comment:
This could be a `zend_closure` flag, passed to `zend_create_closure()` by `ZEND_DECLARE_LAMBDA_FUNCTION`. The flag can be stored in `OBJ_EXTRA_FLAGS()` as there is no other place to store a flag in `zend_closure`. I do this in the pfa branch: https://github.com/arnaud-lb/php-src/blob/23dce325280c7c6c90043ad3bf830073f6838df3/Zend/zend_types.h#L873 + https://github.com/arnaud-lb/php-src/blob/23dce325280c7c6c90043ad3bf830073f6838df3/Zend/zend_closures.c#L31